### PR TITLE
Aml 828 extend api

### PR DESCRIPTION
**Pull request by dashton82**
Extended the API for the data team to use for gathering account information

### CODE REVIEW

File: src/SFA.DAS.Account.Api.Client/Dtos/AccountInformationViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public class AccountInformationViewModel
+    {
+        public string DasAccountName { get; set; }
+        public DateTime DateRegistered { get; set; }
+        public string OrganisationRegisteredAddress { get; set; }
+        public string OrganisationSource { get; set; }
+        public string OrganisationStatus { get; set; }
+        public string OrganisationName { get; set; }
+        public DateTime OrgansiationCreatedDate { get; set; }
+        public string DasAccoundId { get; set; }
```
Comment (robinoconnor) [inline src/SFA.DAS.Account.Api.Client/Dtos/AccountInformationViewModel.cs]: AccoudId should be AccountId

File: src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using System.Web.Http;
+using MediatR;
+using NLog;
+using SFA.DAS.EAS.Api.Models;
+using SFA.DAS.EAS.Application;
+using SFA.DAS.EAS.Application.Queries.GetPagedEmployerAccountsByDateRange;
+
+namespace SFA.DAS.EAS.Api.Controllers
+{
+    [RoutePrefix("api/accountsinformation")]
```
Comment (robinoconnor) [inline src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs]: Would api/accountsinformation not be api/accounts/information

File: src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                {
+                    DasAccountName = result.DasAccountName,
+                    DateRegistered = result.DateRegistered,
+                    OrganisationRegisteredAddress = result.OrganisationRegisteredAddress,
+                    OrganisationSource = result.OrganisationSource,
+                    OrganisationStatus = result.OrganisationStatus,
+                    OrganisationName = result.OrganisationName,
+                    OwnerEmail = result.OwnerEmail,
+                    OrgansiationCreatedDate = result.OrgansiationCreatedDate,
+                    DasAccoundId = result.DasAccoundId
```
Comment (robinoconnor) [inline src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs]: AccoudId to be AccountId (resharper will likely just rename this once the model has changed)

File: src/SFA.DAS.EAS.Api/Models/AccountInformationViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        public string DasAccountName { get; set; }
+        public DateTime DateRegistered { get; set; }
+        public string OrganisationRegisteredAddress { get; set; }
+        public string OrganisationSource { get; set; }
+        public string OrganisationStatus { get; set; }
+        public string OrganisationName { get; set; }
+        public DateTime OrgansiationCreatedDate { get; set; }
+        public string OwnerEmail { get; set; }
+        public string DasAccoundId { get; set; }
```
Comment (robinoconnor) [inline src/SFA.DAS.EAS.Api/Models/AccountInformationViewModel.cs]: AccoundId to be changed to AccountId

File: src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Queries/GetPagedEmployerAccountsByDateRangeTests/WhenIValidateTheQuery.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        [Test]
+        public void ThenFalseIsReturnedWhenTheFieldsArentPopulated()
+        {
+            //Act
+            var actual =_validator.Validate(new GetPagedEmployerAccountsByDateRangeQuery());
+
+            //Assert
+            Assert.IsNotNull(actual);
+            Assert.IsFalse(actual.IsValid());
+            Assert.Contains(new KeyValuePair<string,string>("PageNumber", "PageNumber has not been supplied"),actual.ValidationDictionary );
```
Comment (robinoconnor) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Queries/GetPagedEmployerAccountsByDateRangeTests/WhenIValidateTheQuery.cs]: Are the page number and page size not defaulted if they are not set?

File: src/SFA.DAS.EmployerApprenticeshipsService.Domain/Entities/Account/AccountInformation.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        public string DasAccountName { get; set; }
+        public DateTime DateRegistered { get; set; }
+        public string OrganisationName { get; set; }
+        public string OrganisationRegisteredAddress { get; set; }
+        public string OrganisationStatus { get; set; }
+        public string OwnerEmail { get; set; }
+        public string OrganisationSource { get; set; }
+        public DateTime OrgansiationCreatedDate { get; set; }
+        public string DasAccoundId { get; set; }
```
Comment (robinoconnor) [inline src/SFA.DAS.EmployerApprenticeshipsService.Domain/Entities/Account/AccountInformation.cs]: AccoundId to be changed to AccountId

File: src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Queries/GetPagedEmployerAccountsByDateRangeTests/WhenIValidateTheQuery.cs (hunk 5 above)
Comment (dashton82) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Queries/GetPagedEmployerAccountsByDateRangeTests/WhenIValidateTheQuery.cs]: Depends on implementation of the Query

File: src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs (hunk 2 above)
Comment (dashton82) [inline src/SFA.DAS.EAS.Api/Controllers/EmployerAccountsInformationController.cs]: its intended to be a seperate controller

**Pull request merged**